src/pixelator/pixeldataset/precomputed_layouts.py

```diff
@@ -87,9 +87,16 @@ def partitioning(self) -> list[str]:
         """
         return self._partitioning
 
-    def to_df(self) -> pd.DataFrame:
-        """Return the layouts as a pandas DataFrame."""
-        return self.lazy.collect().to_pandas()
+    def to_df(self, columns: list[str] | None = None) -> pd.DataFrame:
+        """Return the layouts as a pandas DataFrame.
+
+        :param columns: the columns to return, if `None` all columns will be returned
```

Comment:
I see how this works, but it is a bit counter-intiutive. Good that is documented!